include/envoy/network/dns.h

```diff
@@ -39,7 +39,8 @@ class DnsResolver {
    * @return if non-null, a handle that can be used to cancel the resolution.
    *         This is only valid until the invocation of callback or ~DnsResolver().
    */
-  virtual ActiveDnsQuery* resolve(const std::string& dns_name, ResolveCb callback) PURE;
+  virtual ActiveDnsQuery* resolve(const std::string& dns_name, uint32_t port,
```

Comment:
I think I would favor a new method resolveSrv(...). Along with this, a new ResolveSrvCb, which returns a new SrvAddress interface which contains an Address, along with other info like weight, priority, etc. This could then be mapped into a HostImpl construction by the caller where appropriate. 

I'm not sure that we really need "srv_options" right now vs. just doing something like I said above such as "tcp://foo.service:srv" or variant, but I don't feel super strongly.